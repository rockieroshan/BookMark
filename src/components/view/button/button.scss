@import '../../../assets/scss/main.scss';

.btnWrapper {
   @include flexLeftAlignedHorizontally;

    .deleteBtn {
        color: white;
        border: none;
        background: $blackColor;
        padding: 1em 2em;
        cursor: pointer;
        border-radius: 4px;
        margin: 0 auto;
    }
    
    .addBtn {
        @extend .deleteBtn;
        background: $buttonColor;
        margin: 1em 0;
    }

    @include screenType(tablet) {
        justify-content: center;
    }
}