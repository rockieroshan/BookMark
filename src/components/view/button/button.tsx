import { Component, h, Prop, State } from '@stencil/core';

@Component({
    tag: 'app-button',
    styleUrl: 'button.scss',
    shadow: true,
})

export class Button {
    @Prop() isDeleteButton: boolean;
    @Prop() clickHandler: () => void;
    @State() text: string;

    render() {
        const text = this.isDeleteButton ? 'Delete' : 'Add';
        return (
            <div class='btnWrapper'>
                <button class={this.isDeleteButton ? 'deleteBtn' : 'addBtn'} onClick={this.clickHandler}>{text}</button>
            </div>
        );
    }
}
