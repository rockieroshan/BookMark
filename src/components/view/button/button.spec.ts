import { newSpecPage } from '@stencil/core/testing';
import { Button } from './button';


describe('Button', () => {

    it('should render Button', async () => {

        const page = await newSpecPage({
            components: [Button],
            html: `<app-button></app-button>`,
        });

        expect(page.root).toEqualHtml(`
    <app-button>
        <mock:shadow-root>
            <div class="btnWrapper">
            <button class="addBtn">
             Add
          </button>
        </div>
       </mock:shadow-root>
     </app-button>
  `);

        expect(page).toBeTruthy();

        expect(page.root).toMatchSnapshot()
    });

});