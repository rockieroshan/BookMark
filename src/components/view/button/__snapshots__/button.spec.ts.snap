// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Button 1`] = `
<app-button>
  <mock:shadow-root>
    <div class="btnWrapper">
      <button class="addBtn">
        Add
      </button>
    </div>
  </mock:shadow-root>
</app-button>
`;

exports[`should render Button 1`] = `
<app-button>
  <mock:shadow-root>
    <div class="btnWrapper">
      <button class="addBtn">
        Add
      </button>
    </div>
  </mock:shadow-root>
</app-button>
`;
